engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala

```diff
@@ -142,29 +152,67 @@ object CromwellRuntimeAttributes {
       case scala.util.Success(x) => x.to(MemoryUnit.GB).amount.successNel
       case scala.util.Failure(x) => x.getMessage.failureNel
     }
+    val walltime = validateWalltime(attributeMap.get(WALLTIME))
+    val queue = validateQueue(attributeMap.get(QUEUE))
     val bootDiskSize = validateBootDisk(attributeMap.get(BOOT_DISK))
-
-    (docker |@| zones |@| failOnStderr |@| continueOnReturnCode |@| cpu |@| preemptible |@| disks |@| memory |@| bootDiskSize) {
-      new CromwellRuntimeAttributes(attributes, _, _, _, _, _, _, _, _, _)
-    } match {
+    
+    /*
+     * scalaz.syntax.ApplicativeBuilder |@| (as of version 7.3, anyhow) runs out of steam at 12
+     * arguments so we can't just keep extending what Broad had here to add queue, walltime, etc.
+     * The chained <*> below basically does the same thing.
+     */
+    val curriedCtor = (CromwellRuntimeAttributes(_: Map[String, WdlValue], _: Option[String], _: Seq[String], _: Boolean, _: ContinueOnReturnCode, _: Int, _: Int, _: Seq[JesAttachedDisk], _: Double, _: String, _: Option[String], _: Int)).curried
+    bootDiskSize <*> (queue <*> ( walltime <*> (memory <*> ( disks <*> ( preemptible <*> ( cpu <*> (continueOnReturnCode <*> ( failOnStderr <*> ( zones <*> (docker map curriedCtor(attributes))))))))))) match {
```

Comment:
Haha I do enjoy looking at such things!  But also scared... I'm kind of leery of things like shapeless and even much of scalaz that might make my code look like magic to six-months-later-me, let alone poor non-me saps who have to maintain it. I'll leave that gear to the hard-core typeonauts... in production, anyway :)

Date: Fri, 1 Jul 2016 14:44:43 -0700
From: notifications@github.com
To: cromwell@noreply.github.com
CC: conrad.leonard@hotmail.com; author@noreply.github.com
Subject: Re: [broadinstitute/cromwell] Add PBS backend (#1106)

In engine/src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala:

>      val bootDiskSize = validateBootDisk(attributeMap.get(BOOT_DISK))
> -
> -    (docker |@| zones |@| failOnStderr |@| continueOnReturnCode |@| cpu |@| preemptible |@| disks |@| memory |@| bootDiskSize) {
> -      new CromwellRuntimeAttributes(attributes, _, _, _, _, _, _, _, _, _)
> -    } match {
> +    
> +    /*
> +     * scalaz.syntax.ApplicativeBuilder |@| (as of version 7.3, anyhow) runs out of steam at 12
> +     * arguments so we can't just keep extending what Broad had here to add queue, walltime, etc.
> +     * The chained <*> below basically does the same thing.
> +     */
> +    val curriedCtor = (CromwellRuntimeAttributes(_: Map[String, WdlValue], _: Option[String], _: Seq[String], _: Boolean, _: ContinueOnReturnCode, _: Int, _: Int, _: Seq[JesAttachedDisk], _: Double, _: String, _: Option[String], _: Int)).curried
> +    bootDiskSize <*> (queue <*> ( walltime <*> (memory <*> ( disks <*> ( preemptible <*> ( cpu <*> (continueOnReturnCode <*> ( failOnStderr <*> ( zones <*> (docker map curriedCtor(attributes))))))))))) match {

Not at all required for this PR, but if you enjoy looking at mind-bending type-level code for boilerplate removal, some of this might be useful here someday:


http://meta.plasm.us/posts/2013/06/05/applicative-validation-syntax/


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.